packages/toolpad-utils/src/collections.ts

```diff
@@ -108,3 +108,18 @@ export function equalProperties<P extends object>(obj1: P, obj2: P, subset?: (ke
   }
   return true;
 }
+
+/**
+ * Sorts an array of objects by the value of one property.
+ */
+export function sortBy<P extends object>(objs: P[], prop: keyof P, isAsc = true): P[] {
+  return objs.sort((a, b) => {
```

Comment:
None of the other exports of this file mutate the source array. Feels a bit counterintuitive, and I feel like it could lead to bugs. An alternative could be to add a 

```tsx
export function createPropertyComparator<T>(propName: string): (a: T, b: T) => number;
```

and let the consumer decide whether they want to mutate their array or not.

Reply:
I think it's ok to just make this method immutable, simpler and more intuitive. At first I guess I wanted it to be more like `.sort`, but the way that we use it it's probably better for it just to be immutable.